File: Algorithms/Java/Maximum Subarray.java
public class Solution {
    public int maxSubArray(int[] A) {
        int Max = A[0];
		for (int i = 0; i < A.length; i++) {
			if (Max < A[i])
				Max = A[i];
		}
		int sum = 0;
		for (int i : A) {
			if (sum > 0) {
				sum += i;
				if (sum > Max)
					Max = sum;
				continue;
			}
			if (i >= 0) {
				sum = 0;
				sum += i;
			}
		}
		return Max;
    }
}

File: Algorithms/Python/Maximum-Subarray.py
class Solution(object):

    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans, temp = nums[0], nums[0]
        for x in nums[1:]:
            if temp < 0:
                if x >= 0:
                    temp = x
            else:
                temp += x
            ans = max([ans, temp, x])
        return ans
